packages/medusa/src/strategies/tax-calculation.ts

```diff
@@ -58,8 +67,15 @@ class TaxCalculationStrategy implements ITaxCalculationStrategy {
     for (const sm of shipping_methods) {
       const amount = sm.price
       const lineRates = taxLines.filter((tl) => tl.shipping_method_id === sm.id)
+
       for (const lineRate of lineRates) {
-        taxTotal += Math.round(amount * (lineRate.rate / 100))
+        const rate = lineRate.rate / 100
+
+        taxTotal += calculatePriceTaxAmount({
+          price: amount,
+          taxRate: rate,
+          includesTax: sm.includes_tax,
```

Comment:
suggestion: use flag router here

Reply:
As you are changing the shipping methods in this particular file, I will discard the changes I've done.